tmt/utils.py

```diff
@@ -2010,6 +2010,30 @@ def public_git_url(url: str) -> str:
     return url
 
 
+def public_web_url(fmf_id: 'tmt.base.FmfId', path: str) -> str:
+    """
+    Convert a public git url into a clickable web url format
+
+    Compose a clickable link from git url, ref and path to file
+    for most common git servers
+    """
+    assert fmf_id.url is not None
+    assert fmf_id.ref is not None
+    assert fmf_id.path is not None
```

Comment:
Is it guaranteed all of these cannot be `None` when `public_web_url` is called? I'm looking at `Test.fmf_id`, and it *may* leave `path` unset, for example. What guarantees all these fields have valid values? Please, add a comment mentioning what makes the use of `assert` safe here, for other humans reading the code in the future :)

Reply:
yeah, thanks, I'm growing increasingly hateful of this mypy and then I make these bugs like fmf_id.path really can be None and most likely is for 99% of cases

Reply:
> I'm growing increasingly hateful of this mypy

I'm sorry to hear that...